src/main/kotlin/org/jetbrains/research/testspark/helpers/psi/kotlin/KotlinPsiHelper.kt

```diff
@@ -0,0 +1,200 @@
+import org.jetbrains.kotlin.resolve.lazy.BodyResolveMode
+import org.jetbrains.research.testspark.helpers.psi.Language
+import org.jetbrains.research.testspark.helpers.psi.PsiClassWrapper
+import org.jetbrains.research.testspark.helpers.psi.PsiHelper
+import org.jetbrains.research.testspark.helpers.psi.PsiMethodWrapper
+import org.jetbrains.research.testspark.tools.llm.SettingsArguments
+
+class KotlinPsiHelper(private val psiFile: PsiFile) : PsiHelper {
+
+    override val language: Language get() = Language.Kotlin
+
+    private val log = Logger.getInstance(this::class.java)
+
+    override fun generateMethodDescriptor(psiMethod: PsiMethodWrapper): String {
+        val methodDescriptor = psiMethod.methodDescriptor
+        log.info("Method description: $methodDescriptor")
+        return methodDescriptor
+    }
+
+    override fun getSurroundingClass(caretOffset: Int): PsiClassWrapper? {
+        val element = psiFile.findElementAt(caretOffset)
+        val cls = element?.parentOfType<KtClassOrObject>(withSelf = true)
+
+        if (cls != null) {
+            if (cls.name != null) {
+                val kotlinClassWrapper = KotlinPsiClassWrapper(cls)
+                log.info("Surrounding class for caret in $caretOffset is ${kotlinClassWrapper.qualifiedName}")
+                return kotlinClassWrapper
+            }
+        }
+
+        log.info("No surrounding class for caret in $caretOffset")
+        return null
+    }
+
+    override fun getSurroundingMethod(caretOffset: Int): PsiMethodWrapper? {
+        val element = psiFile.findElementAt(caretOffset)
+        val method = element?.parentOfType<KtFunction>(withSelf = true)
+
+        if (method != null) {
+            val wrappedMethod = KotlinPsiMethodWrapper(method)
+            log.info("Surrounding method for caret at $caretOffset is ${wrappedMethod.methodDescriptor}")
+            return wrappedMethod
+        }
+
+        log.info("No surrounding method for caret at $caretOffset")
+        return null
+    }
+
+    override fun getSurroundingLine(caretOffset: Int): Int? {
+        val doc = PsiDocumentManager.getInstance(psiFile.project).getDocument(psiFile) ?: return null
+
+        val selectedLine = doc.getLineNumber(caretOffset)
+        val selectedLineText =
+            doc.getText(TextRange(doc.getLineStartOffset(selectedLine), doc.getLineEndOffset(selectedLine)))
+
+        if (selectedLineText.isBlank()) {
+            log.info("Line $selectedLine at caret $caretOffset is not valid")
+            return null
+        }
+        log.info("Surrounding line at caret $caretOffset is $selectedLine")
+        return selectedLine
+    }
+
+    override fun collectClassesToTest(
+        project: Project,
+        classesToTest: MutableList<PsiClassWrapper>,
+        caretOffset: Int,
+    ) {
+        val maxPolymorphismDepth = SettingsArguments(project).maxPolyDepth(0)
+        val cutPsiClass = getSurroundingClass(caretOffset)!!
```

Comment:
Re-write the `.maxPolyDepth(0)` with a named method parameter please, it increases clarity.